UndertaleModLib/Decompiler/Instructions/Decompiler.AssignmentStatement.cs

```diff
@@ -5,40 +5,54 @@ namespace UndertaleModLib.Decompiler;
 
 public static partial class Decompiler
 {
-    // Represents a high-level assignment statement.
+    /// <summary>
+    /// Represents a high-level assignment statement.
+    /// </summary>
     public class AssignmentStatement : Statement
     {
+        /// <summary>
+        /// The variable that will get something assigned to.
+        /// </summary>
         public ExpressionVar Destination;
+        /// <summary>
+        /// The value that will be assigned.
+        /// </summary>
         public Expression Value;
 
-        public bool HasVarKeyword;
+        /// <summary>
+        /// Whether the destination variable is a local variable.
+        /// </summary>
+        public bool IsLocalVariable;
 
         private bool _isStructDefinition, _checkedForDefinition;
+        
+        /// <summary>
+        /// Whether the destination variable is a struct.
+        /// </summary>
         public bool IsStructDefinition
         {
             get
             {
                 // Quick hack
-                if (!_checkedForDefinition)
+                if (_checkedForDefinition) return _isStructDefinition;
```

Comment:
I'm not a huge fan of the single-line formatting, but I won't oppose it. Up to you.

Reply:
I completely forgot to respond to this.
I did want to change it. will do tomorrow.